### PR TITLE
Spring Data Jpa

### PR DESCRIPTION
**Pull request by VlasenkoYevhenii**

### CODE REVIEW

File: src/main/java/com/example/bookstoreapplication/controller/BookController.java [hunk 1]
```diff
@@ -33,4 +37,15 @@ public BookDto getBookById(@PathVariable Long id) {
     public BookDto createBook(@Valid @RequestBody CreateBookRequestDto dto) {
         return bookService.save(dto);
     }
+
+    @PutMapping("/{id}")
```
Comment (k0sm0naft) [inline src/main/java/com/example/bookstoreapplication/controller/BookController.java:41]: You can add response status.

File: src/main/resources/db/changelog/changes/01-create-book-table.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        - createTable:
+            tableName: books
+            columns:
+              - column:
+                  name: id
+                  type: BIGINT
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+                    nullable: false
```
Comment (k0sm0naft) [inline src/main/resources/db/changelog/changes/01-create-book-table.yaml:15]: Primary Key by default is NOT NULL
```suggestion
```

File: src/main/resources/db/changelog/changes/01-create-book-table.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+              - column:
+                  name: title
+                  type: VARCHAR(255)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: author
+                  type: VARCHAR(255)
+                  constraints:
+                    unique: true
```
Comment (k0sm0naft) [inline src/main/resources/db/changelog/changes/01-create-book-table.yaml:25]: Why unique? If author write few books, we can't add them.
Better add `nullable: false` here.

File: src/main/resources/db/changelog/changes/01-create-book-table.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
+databaseChangeLog:
```
Comment (k0sm0naft) [inline src/main/resources/db/changelog/changes/01-create-book-table.yaml:1]: If you add constrains, batter, give them names.

File: src/main/java/com/example/bookstoreapplication/service/BookServiceImpl.java [hunk 5]
```diff
@@ -31,7 +31,19 @@ public List<BookDto> findAll() {
...
     @Override
     public BookDto getById(Long id) {
-        return mapper.toDto(bookRepository.getById(id)
+        return mapper.toDto(bookRepository.findById(id)
                 .orElseThrow(() -> new EntityNotFoundException("Failed to get book by id=" + id)));
     }
+
+    @Override
+    public BookDto updateBook(Long id, CreateBookRequestDto dto) {
+        Book book = mapper.toModel(dto);
```
Comment (k0sm0naft) [inline src/main/java/com/example/bookstoreapplication/service/BookServiceImpl.java:40]: You cant add or update book with isbn that already soft-deleted.

**Review by k0sm0naft — CHANGES_REQUESTED** [review]
Needed few changes.

File: src/main/java/com/example/bookstoreapplication/service/BookServiceImpl.java (hunk 5 above)
Comment (VlasenkoYevhenii) [reply src/main/java/com/example/bookstoreapplication/service/BookServiceImpl.java:40]: yep, and it won't.

File: src/main/resources/db/changelog/changes/01-create-book-table.yaml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+                  type: VARCHAR(255)
+              - column:
+                  name: cover_image
+                  type: VARCHAR(255)
+              - column:
+                  name: is_deleted
+                  type: boolean
+                  defaultValueBoolean: false
+                  constraints:
+                    nullable: false
```
Comment (andrii-hoienko) [inline src/main/resources/db/changelog/changes/01-create-book-table.yaml:47]:
```suggestion
                    nullable: false
                    
```

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 7]
```diff
@@ -0,0 +1,3 @@
+databaseChangeLog:
+  - include:
+      file: db/changelog/changes/01-create-book-table.yaml
```
Comment (andrii-hoienko) [inline src/main/resources/db/changelog/db.changelog-master.yaml:3]:
```suggestion
      file: db/changelog/changes/01-create-book-table.yaml
      
```

**Review by andrii-hoienko — APPROVED** [review]
Good job!

**Pull request merged**